src/eventgrid/azext_eventgrid/_params.py

```diff
@@ -84,8 +160,31 @@ def load_arguments(self, _):    # pylint: disable=too-many-statements
         c.argument('odata_query', arg_type=odata_query_type)
         c.argument('domain_name', arg_type=domain_name_type)
         c.argument('domain_topic_name', arg_type=domain_topic_name_type)
-        c.argument('allow_traffic_from_all_ips', arg_type=get_three_state_flag(), options_list=['--allow-traffic-from-all-ips'], help="Allow traffic from all IPs.")
+        c.argument('system_topic_name', arg_type=system_topic_name_type)
+        c.argument('source', help="The ARM Id for the topic, e.g., /subscriptions/{SubId}/resourceGroups/{RgName}/providers/Microsoft.Storage/storageAccounts/{AccountName}")
+        c.argument('private_endpoint_connection_name', arg_type=private_endpoint_connection_name_type)
+        c.argument('private_link_resource_name', arg_type=private_link_resource_name_type)
+        c.argument('public_network_access', arg_type=public_network_access_type)
         c.argument('inbound_ip_rules', action=AddInboundIpRule, nargs='+')
+        c.argument('sku', arg_type=sku_type)
+        c.argument('identity', arg_type=identity_type)
```

Comment:
suggest to implement msi as command for consistent across all az cli commands, reference interface:
vm: https://docs.microsoft.com/en-us/cli/azure/vm/identity?view=azure-cli-latest. webapp: https://docs.microsoft.com/en-us/cli/azure/webapp/identity?view=azure-cli-latest

Reply:
Thanks for the feedback. This will require additional support of functionalities from our RP. We are evaluating this in our team and if needed will fix them as bug fix for this cli extension (still in preview) as it will requires swagger changes and python sdk regeneration. This should not block this PR.

Reply:
thanks for detail explanation. pls create an new issue to track the change, since azure cli is working on change msi interface to make it consistent.